### ISSUE TITLE
Should we deprecate HAL+JSON for 2.x?

### ISSUE THREAD

**Issue by e0ipso** [RESTful-Drupal/restful#601]
For me, everything we support should be SimpleJson and JsonApi.

I'm thinking that we may move HAL+JSON to a sub-module or contrib module.

@amitaibu, thoughts?

**Comment by thebruce** [comment]
First: You guys are kings among men and this is your baby.

Additionally, @mateu-aguilo-bosch you didn't ask me -- but -- 

two things going for keeping hal+json:
1) It is a very good HATEOAS based read only format

2) SimpleJson, JsonApi and hal+json satisfies The "rule of threes" from Will Tracz  (Confessions of a Used Program Salesman, Addison-Wesley, 1995) for your plug-able formats right within your module.

That precept is:

"A reusable component should be tried out in three different applications before it will be sufficiently general to accept into a reuse library"

That being said, it's more work, I know what that represents.  Moving it to contrib or sub module - doesn't preclude the above concept.

**Comment by amitaibu** [comment]
Yeah, I'm totally ok with moving HAL to contrib. I know who can be an awesome maintainer of it (hint: it's @thebruce :wink: )
